### PR TITLE
Feature/fix flat conformance test

### PR DESCRIPTION
**Pull request by stefanspiska**
Added Test and fixes for https://github.com/better-care/web-template-tests

- DoraBuilderTest
-  DrpBuilderTest
- EthercisBuilderTest
- FeederAuditTest
- IsmTransitionTest
- NamedElementsTest
- NormalRangeTest
- NullFlavorTest
- TermMappingTest

added test and fix for https://github.com/ehrbase/ehrbase/issues/659

intermediate  merge for https://jira.vitagroup.ag/browse/PEM-432

### CODE REVIEW

File: serialisation_conformance_test/src/test/java/org/ehrbase/webtemplate/CompositionConverterImp.java [hunk 1]
```diff
@@ -111,18 +111,97 @@ public String convertFlatToRaw(
...
         .forEach(e -> currentValues.replace(e.getKey(), "video/H263-2000"));
 
-    // add missing terminology for medra_classification
-    if (currentValues.containsKey(
-        "drug_related_problem_report/medication_error/medra_classification|code")) {
-      currentValues.put(
-          "drug_related_problem_report/medication_error/medra_classification|terminology",
-          "terminology");
-    }
+    // add missing terminology for some flat examples
```
Comment (subigre) [inline serialisation_conformance_test/src/test/java/org/ehrbase/webtemplate/CompositionConverterImp.java:114]: @stefanspiska I am not sure that I fully understand. Why do we need to add the terminology and why it is not included in the example from better? They derive automatically from the template?

Comment (stefanspiska) [reply serialisation_conformance_test/src/test/java/org/ehrbase/webtemplate/CompositionConverterImp.java:114]: The example file from better are simply wrong. Terminology is required and should be path of the flat file.

Comment (stefanspiska) [reply serialisation_conformance_test/src/test/java/org/ehrbase/webtemplate/CompositionConverterImp.java:114]: I did not check if there would set a default if missing.

File: serialisation/src/main/java/org/ehrbase/serialisation/flatencoding/std/marshal/postprocessor/DvOrderedPostprocessor.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    if (rmObject.getNormalRange() != null && rmObject.getNormalRange().getLower() != null) {
+      DvOrdered lower = rmObject.getNormalRange().getLower();
+
+      values.putAll(
+          ((StdConfig) findStdConfig(lower.getClass()))
+              .buildChildValues(term + "/_normal_range/lower", lower, context));
+    }
+
+    if (rmObject.getNormalRange() != null && rmObject.getNormalRange().getUpper() != null) {
+      DvOrdered lower = rmObject.getNormalRange().getUpper();
```
Comment (subigre) [inline serialisation/src/main/java/org/ehrbase/serialisation/flatencoding/std/marshal/postprocessor/DvOrderedPostprocessor.java:49]: Rename `lower` to `upper`

File: serialisation/src/main/java/org/ehrbase/serialisation/flatencoding/std/umarshal/StdToCompositionWalker.java [hunk 3]
```diff
@@ -125,6 +114,26 @@ public void walk(
     }
   }
 
+  public static <T> String buildNamePathWithElementHandling(Context<T> context) {
```
[the remark this answers is not in the document]
Comment (stefanspiska) [reply serialisation/src/main/java/org/ehrbase/serialisation/flatencoding/std/umarshal/StdToCompositionWalker.java:117]: No this is only used for the flat format

**Review by subigre — APPROVED** [review]

**Pull request merged**